virtualbox_snapshotter.py

```diff
@@ -236,29 +267,30 @@ def main():
     """
     logger.info("Starting autosnapshotter script ...")
 
-    if args.list:
-        _, snapshot_details = load_vm_and_snapshot_details(args.machine_name)
-        # Avoiding to use logger here to not clutter output which may be of some use for user
-        print(f"Available snapshots for '{args.machine_name}':")
-        if not snapshot_details:
-            print(f"0 snapshots")
-        else:
-            for snapshot in snapshot_details:
-                print(f"Name: '{snapshot[1]}' UUID: {snapshot[0]}")
-                print(f"\tDescription: {snapshot[2]}")
-            return
+    virtual_machine = load_virtual_machine(args.machine_name)
+    snapshot_details = load_snapshot_details(virtual_machine)
+
+    # All manipulations with virtual machine will happen under same session
+    session = virtualbox.Session()
 
+    if args.list:
+        list_snapshots(args.machine_name, snapshot_details)
+        # Always exit after running with `-l`/`--list` argument
+        sys.exit(0)
```

Comment:
Now it returns all the time after printing snapshot details or printing that no snapshots are available